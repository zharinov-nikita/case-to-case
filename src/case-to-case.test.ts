import { describe, expect, test } from '@jest/globals'
import { CASE } from './case-to-case.constant'
import { CaseToCase } from './case-to-case.module'
import type { TypeDefineCase, TypeIsCase, TypeToCase } from './case-to-case.type'

describe('caseToCase', () => {
  const caseToCase = new CaseToCase()

  const caseValues = Object.values(CASE)
  const caseKeys = Object.keys(CASE) as TypeDefineCase[]

  function generateMethodName(value: TypeDefineCase, prefix: 'is' | 'to'): TypeIsCase | TypeToCase {
    return (prefix + value.charAt(0).toUpperCase() + value.slice(1)) as TypeIsCase | TypeToCase
  }

  caseKeys.forEach((caseKey, index) => {
    const methodName = generateMethodName(caseKey, 'to')
    const stringInAnyCase = caseValues[index]
    const toBe = caseValues[index]

    test(methodName, () => {
      expect(caseToCase[methodName](stringInAnyCase)).toBe(toBe)
      expect(caseToCase[methodName](stringInAnyCase)).toMatchSnapshot()
    })
  })

  caseKeys.forEach((caseKey, index) => {
    const methodName = generateMethodName(caseKey, 'is')
    const stringInAnyCase = caseValues[index]
    const toBe = caseToCase[methodName](stringInAnyCase) === caseToCase[methodName](stringInAnyCase)

    test(methodName, () => {
      expect(caseToCase[methodName](stringInAnyCase)).toBe(toBe)
      expect(caseToCase[methodName](stringInAnyCase)).toMatchSnapshot()
    })
  })
})
