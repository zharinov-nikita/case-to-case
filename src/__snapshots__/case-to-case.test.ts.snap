// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`caseToCase isCamelCase 1`] = `true`;

exports[`caseToCase isCapitalizeCase 1`] = `true`;

exports[`caseToCase isLowerCase 1`] = `true`;

exports[`caseToCase isLowerKebabCase 1`] = `true`;

exports[`caseToCase isLowerSnakeCase 1`] = `true`;

exports[`caseToCase isPascalCase 1`] = `true`;

exports[`caseToCase isUpperCase 1`] = `true`;

exports[`caseToCase isUpperKebabCase 1`] = `true`;

exports[`caseToCase isUpperSnakeCase 1`] = `true`;

exports[`caseToCase toCamelCase 1`] = `"camelCase"`;

exports[`caseToCase toCapitalizeCase 1`] = `"Capitalizecase"`;

exports[`caseToCase toLowerCase 1`] = `"lowercase"`;

exports[`caseToCase toLowerKebabCase 1`] = `"lower-kebab-case"`;

exports[`caseToCase toLowerSnakeCase 1`] = `"lower_snake_case"`;

exports[`caseToCase toPascalCase 1`] = `"PascalCase"`;

exports[`caseToCase toUpperCase 1`] = `"UPPERCASE"`;

exports[`caseToCase toUpperKebabCase 1`] = `"UPPER-KEBAB-CASE"`;

exports[`caseToCase toUpperSnakeCase 1`] = `"UPPER_SNAKE_CASE"`;
